README.md
# [![das-js](https://img.shields.io/endpoint?url=https://dashboard.cypress.io/badge/detailed/qnmnp9/master&style=flat&logo=cypress)](https://dashboard.cypress.io/projects/qnmnp9/runs)

![Gatsby Build](https://github.com/ragaeeb/das-js/workflows/Gatsby%20Build/badge.svg)
![Tests](https://github.com/ragaeeb/das-js/workflows/Tests/badge.svg)
![Deploy release](https://github.com/ragaeeb/das-js/workflows/Deploy%20release/badge.svg)
![Check Markdown links](https://github.com/ragaeeb/das-js/workflows/Check%20Markdown%20links/badge.svg)
[![GitHub](https://img.shields.io/github/license/ragaeeb/das-js?color=blue)](https://github.com/ragaeeb/das-js/blob/master/LICENSE.md) ![GitHub stars](https://img.shields.io/github/stars/ragaeeb/das-js) ![GitHub forks](https://img.shields.io/github/forks/ragaeeb/das-js) ![GitHub issues](https://img.shields.io/github/issues/ragaeeb/das-js)

## A clean, beautiful and responsive template for masājid and Islāmic centres.

<h2>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/example.gif" alt="Islamic Simplefolio" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/monthly.jpg" alt="Monthly Schedule" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/yearly.jpg" alt="Yearly Schedule" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/all_posts.png" alt="All Posts" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/sample_post.jpg" alt="Sample Post" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/mailing_list.jpg" alt="Newsletter" width="600px" />
  <br>
  <img src="https://github.com/ragaeeb/das-js/blob/master/examples/privacy.jpg" alt="Privacy Policy" width="600px" />
</h2>

## Features

⚡️ Modern UI Design + Reveal Animations\
⚡️ One Page Layout built with `React`, exported as `Preact` for optimization\
⚡️ Styled with Bootstrap v4.3 + Custom SCSS\
⚡️ Fully Responsive\
⚡️ Lazy Loading for Better SEO\
⚡️ Image optimization with Gatsby\
⚡️ Accurate and dynamic prayer times with support for dynamic iqāmah times\
⚡️ Ability to view upcoming and previous prayer times.
⚡️ Hijri date display\
⚡️ Automatic generation of monthly prayer timetable\
⚡️ Beautiful visual graph generation of annual prayer timings\
⚡️ Up to `1000` blog posts\
⚡️ Beautiful visual pie chart of monthly expenses for what your audience will be donating for\
⚡️ Privacy Policy built-in and included\
⚡️ Clear documentation steps on how to replicate for your masjid\
⚡️ Built in dynamic last-update feature with direct-link to commit history\
⚡️ Well unit-tested\
⚡️ Integrated support for 1:1 engagement with audience using `Chatwoot`.

To view a demo example, **[click here](https://dar-as-sahaba.com/)**\

---

## Getting Started 🚀

These instructions will get your Islāmic centre up and running from scratch bootstrapped with `Islamic Simplefolio`.

### Domain name registration

Go to a domain name registration platform and register a domain name. In this example we will use [namecheap.com](https://namecheap.com) and purchase a domain name for your masjid. In this example we will assume we have registered [dar-as-sahaba.com](https://dar-as-sahaba.com). Note that you do not need any of add-ons like SSL, or anything else. However for privacy reasons I suggest you add in the `Free Whoisguard` add-on.

### Cloudflare

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/cloudflare_dns.jpg" alt="Nameserver settings on Cloudflare" />

`Cloudflare` provides websites a set of free services such as `DDOS` protection and other speed optimizations that will be useful. We will be changing our name servers to Cloudflare to let it route our traffic.

1. Go to [Cloudflare](https://cloudflare.com), create an account, and then `Add a Site`.
2. Choose the `FREE` option.
3. Add the domain-name you have just registered (ie: [dar-as-sahaba.com](https://dar-as-sahaba.com))
4. Once it asks you to validate all the DNS information, simply continue.
5. Once everything is complete, `Cloudflare` will give you some name servers you need to replace your existing ones on `Namecheap` with.
6. Open another tab on your browser and go back to Namecheap, and click on `Manage` to edit your domain name settings.
7. Under your domain, in the `Domain` tab, you will see a `NAMESERVERS` section, from the dropdown use `Custom DNS`.
8. In the two fields, set the two `something.ns.cloudflare.com` addresses that `Cloudflare` gave you.
9. Save the settings.
10. Go back on `Cloudflare` and click the `Recheck Nameservers` button.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/namecheap_dns.jpg" alt="Nameserver Settings on Namecheap" />

At this point this may take some time (ie: usually an hour or so) to change the nameservers over.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/cloudflare_success.jpg" alt="Cloudflare success message" />

11. Go to the `DNS` section.
12. Add the following `A` records:

```json
[
  {
    "type": "A",
    "name": "@",
    "content": "185.199.111.153",
    "ttl": "Auto",
    "proxyStatus": "Proxied"
  },
  {
    "type": "A",
    "name": "@",
    "content": "185.199.110.153",
    "ttl": "Auto",
    "proxyStatus": "Proxied"
  },
  {
    "type": "A",
    "name": "@",
    "content": "185.199.109.153",
    "ttl": "Auto",
    "proxyStatus": "Proxied"
  },
  {
    "type": "A",
    "name": "@",
    "content": "185.199.108.153",
    "ttl": "Auto",
    "proxyStatus": "Proxied"
  },
  {
    "type": "MX",
    "name": "@",
    "mailServer": "mx1.improvmx.com.",
    "priority": "10",
    "ttl": "Auto"
  },
  {
    "type": "MX",
    "name": "@",
    "mailServer": "mx2.improvmx.com.",
    "priority": "20",
    "ttl": "Auto"
  },
  {
    "type": "TXT",
    "name": "@",
    "content": "v=spf1 include:spf.efwd.registrar-servers.com include:spf.improvmx.com ~all",
    "ttl": "Auto"
  }
]
```

13. Go to SSL/TLS, and enable `Full`.
14. Go to `Speed` and enable `Auto Minify` to all: `Javascript`, `CSS`, and `HTML`.
15. Disable `Brotli` compression.
16. Enable `Rocket Loader`.
17. Go to `Page Rules`.
18. Click `Create Page Rule`.
19. For `URL matches` set it to: `http://yourdomain.com/*`, in the setting select `Always Use HTTPS`. Click `Save`.
20. Click `Create Page Rule`.
21. For `URL matches` set it to: `www.yourdomain.com/*`, in the setting select `Forwarding URL` with `301 - Permanent Redirect` and set the redirect to `https//yourdomain.com/$1`. Click `Save`.
22. For `URL matches` set it to: `https://www.yourdomain.com/*`, in the setting select `Cache Level` and choose `Cache Everything`. Click `Save`.
23. Enable all the rules.

### Technical Part (Cloning & Configuration)

This is the slightly technical part. But you won't have to download any tools, but you will have to create a few accounts to get everything to work properly.

#### Placeholders

Go to `/src/mock/data.js` and fill your information that are specific for your centre:

##### Hero Section

This section modifies what appears at the very top of the page such as prayer timings, and hijri dates and links.

###### Calculation Parameters

These are configured using `12` degrees for the Fajr and ʿIshāʾ twilight angles (ie: Nautical Twilight) for the most accurate version of calculating the `Fajr` time instead of using the other conventions. Note that using other conventions that your region and other masājid in your area may be using may not result in the most accurate time which is vitally important because it may during the month of Ramaḍān eating Suḥūr later than the allowed time, or praying Fajr before its actual entrance! [Read more about this here](https://ia802902.us.archive.org/32/items/fajr-letter-das/Letter_Fajr-Ishaa_Ottawa-2016-06-27.pdf). Change these at your own discretion.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/geo.jpg" alt="Getting latitude and longitude for an address" />

The only values you should need to modify here are `latitude`, `longitude` and replace those numbers with the exact geographical coordinates for [your Islāmic centre](https://www.latlong.net/convert-address-to-lat-long.html). You will also need to replace the `timeZone` which you can find [here](http://www.timezoneconverter.com/cgi-bin/findzone).

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/timezone.png" alt="Getting a timezone for your location" />

```javascript
export const heroData = {
  calculation: {
    fajrAngle: 12,
    ishaAngle: 12,
    latitude: '45.3506',
    longitude: '-75.793',
    method: 'NauticalTwilight',
    timeZone: 'America/Toronto',
  },
  …
```

###### Labels

If you want to customize the labels of each of the prayer times and events you can do so here.

```javascript
  …
  labels: {
    fajr: 'Fajr',
    sunrise: 'Sunrise',
    dhuhr: 'Dhuhr',
    asr: 'ʿAṣr',
    maghrib: 'Maġrib',
    isha: 'ʿIshāʾ',
    middleOfTheNight: '1/2 Night Begins',
    lastThirdOfTheNight: 'Last 1/3 Night Begins',
    jumuah: 'Khuṭbah al-Jumuʿah',
  },
  …
```

###### Iqāmah Schedule

Some masājid have some variable timings for the different prayers that do not remain upon a fixed schedule throughout the year. As a result this section is a bit more complex, to allow some more customization.

For each of the events (ie: `fajr`, `dhuhr`, `jumuah`, etc.) you can customize exactly when the Iqāmah for that prayer is made in one of three ways.

###### Fixed Iqāmah Timings

If your timings for a prayer will always be a fixed pattern, you can just include them in text like this:

```javascript
  …
  iqamahs: {
    …
    fajr: '+20 mins',
    dhuhr: '+5 mins',
    …
  …
```

This means that no matter what time of the year, the iqāmah for Fajr will always be `20 minutes` after the aḏān and the iqāmah for Dhuhr will always be `5 minutes` after the aḏān. You can replace it with any text there and it will show up, it does not have to follow the `+X minutes` pattern, although this is a good convention to use to remain consistent.

###### Multiple Iqāmahs

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/iqamahs.jpg" alt="Setting multiple iqāmah times" />

If you have more than one congregational prayer (ie: you hold a second jamāʿah), you can enter it in a list like this:

```javascript
  …
  iqamahs: {
    …
    fajr: ['4:00 AM', '4:35 AM'],
    dhuhr: ['+5 mins', '+30 mins'],
    …
  …
```

###### Variable Iqāmahs

If you have very specific timings that vary throughout the year, you will have to be a bit more detailed. For the specific prayer that you need variable iqāmah timings for, you will have to list out all the months (ie: 1-12 with `1` for `January`, and `12` for `December`), and then provide the ranges for each.

The following example shows a masjid for which in `January` the Dhuhr prayer iqāmah will be at `12:30 PM`, then in `February`, they will hold two congregational prayers, one at `12:30 PM`, and one at `1:00 PM`.

From `March 1st` (until `March 7th` inclusive), the iqāmah will be at `12:30 PM`, then starting `March 8th` (until `March 20th` inclusive) the iqāmah will be at `1:30 PM`. Finally starting on `March 21st` until the end of the month, there will be two congregational prayers, one at `1:30 PM` and the other at `1:45 PM`.

Then for the rest of the year, the Dhuhr iqāmah will be fixed to `1:30 PM`.

You can add as many ranges as you want.

```javascript
  …
  iqamahs: {
    …
    dhuhr: {
      1: '12:30 PM',
      2: ['12:30 PM', '1:00 PM'],
      3: {
        1: '12:30 PM',
        8: '1:30 PM',
        21: ['1:30 PM', '1:45 PM'],
      },
      4: '1:30 PM',
      5: '1:30 PM',
      6: '1:30 PM',
      7: '1:30 PM',
      8: '1:30 PM',
      9: '1:30 PM',
      10: '1:30 PM',
      11: '12:30 PM',
      12: '12:30 PM',
    },
    …
  …
```

###### Adjusting Hijri Date

Sometimes throughout the year the calculated Hijri date can differ from the Islāmic date due to moonsightings which are supposed to take precedence over the calculated date. This is especially important in months such as `Ramaḍān` where your centre may be following the moonsighting of the locality to determine the Islamic date. During those times you may need to adjust the calculation by modifying the `hijriAdjust` value. `-1` means show the Hijri date to be `1 day before` the calculated one, `-2` means `2 days before`, `0` means leave it exactly as calculated, `1` means show the Hijri date to be `1 day after` the calculated one, `2` means it should be `2 days` after the calculated one, etc.

```javascript
  …
export const heroData = {
  …
  hijriAdjust: -1,
  …
```

###### Links

Sometimes you may need some additional details you want to provide to your audience regarding specific prayers. For example if your `Fajr` prayer timing differs from the other masājid in your locality, you may want to explain why. You can do this by adding the specific prayer you want to add this link for with the URL you want to link that prayer to.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/fajr_link.jpg" alt="Attaching a link to Fajr timing" />

The following example shows both the Fajr and Dhuhr prayer labels that will be rendered as clickable links.

```javascript
  …
export const heroData = {
  …
  links: {
    fajr: 'https://archive.org/download/fajr-letter-das/Letter_Fajr-Ishaa_Ottawa-2016-06-27.pdf',
    dhuhr: 'https://anotherlink.pdf',
  },
  …
```

###### Calendar Link

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/calendar.jpg" alt="Google Calendar link" />

Instead of duplicating your events information on the website, it is optimal for you to create your events on `Google Calendar` and then link to it on the website so your audience can see the most up to date information about it.

```javascript
  …
export const heroData = {
  …
  calendarUrl: 'https://calendar.google.com/calendar/embed?src=60me58mrktt0lt24mijjkhddvc%40group.calendar.google.com&ctz=America%2FNew_York',
  …
```

If you don't want to include a calendar link, simply remove the `calendarUrl` entry from `data.js`.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/calendar_share.jpg" alt="Making a Google Calendar public" />

To create a calendar for your Islāmic centre:

1. Go to [https://calendar.google.com](https://calendar.google.com).
2. On the left-side, under `Other Calendars`, click `Create New Calendar`.
3. Give it a name, description and set the timezone.
4. Click `Create Calendar`.
5. On the left side click on your newly created Calendar.
6. Under `Calendar Settings` scroll down to `Access Permissions`.
7. Check the `Make available to public` and ensure `See all event details` is selected.
8. Scroll down to `Integrate Calendar`, and copy the `Public URL to the calendar` value.
9. This is the URL we will use in the `calendarUrl` to allow your audience to access your centre's events.

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/calendar_url.jpg" alt="Getting the Calendar URL" />

###### Sāʾ al-Istijābah

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/istijaba.jpg" alt="Last Hour on Friday" />

You can display a reminder for all those who visit your website at the time when duʿāʾ is accepted in the last hour of Friday. You can show them a hadith or some custom text in the `istijabaText` field. If you do not want to display anything, then remove that.

```javascript
  …
export const heroData = {
  …
  istijabaText: 'The text you want to show',
  …
```

##### About Section

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/about.jpg" alt="About Us Section" />

This is the information that will be displayed right underneath the updates. You can add as many services as you want by copying and pasting the `title`, and `body` patterns.

```javascript
export const aboutData = {
  paragraphOne: 'First paragraph',
  paragraphTwo: 'Second paragraph',
  services: [
    {
      title: 'Five Daily Ṣalawāt',
      body:
        'The muṣallá is open to allow Muslims (both brothers and sisters) a location to pray their five daily ṣalawāt.',
    },
    {
      title: 'Ṣalat al-Jumʿuah',
      body:
        'Every week, salat-ul-Jumʿuah is held (due to limited space, for brothers only), where the khaṭīb deliver the khuṭbah strictly adhering to the Qur’ān & the Sunnah.',
    },
  ],
};
```

##### Donate Section

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/donate.jpg" alt="Donate Section" />

You can set all the ways your audience can donate to your Islāmic centre here.

For the `PayPal` you can either use a `paypal.me` link, or use a [payment button link](https://www.paypal.com/buttons/).

The `etransfer` email address will automatically be scrambled so that spam bots do not pick it up, but it should show up fine for the users.

In the `expenses` field, you can list out the individual expenses for your Islāmic centre that your audience will be donating to support. The app will automatically add the values up to create a total, and build a pie chart breaking down each expense.

You can remove any of the fields if you do not want to show them.

```javascript
export const donationsData = {
  directDeposit: 'Direct deposit information',
  etransfer: 'youremailaddress@gmail.com',
  expenses: [
    { label: 'Rent', value: 1000 },
    { label: 'Utilities', value: 600 },
  ],
  khutbah: 'After the Friday sermon, donations for the muṣallá are collected.',
  paypal: 'https://paypal.me/yourbrand',
  reminderDawah: 'Reminder text here',
  reminderReward: 'Another reminder about ṣadaqah',
};
```

##### Footer Section

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/footer.png" alt="Footer Section" />

Set your center's social media information here. You can set platforms such as [twitter](https://twitter.com), [soundcloud](https://soundcloud.com), [instagram](https://instagram.com), [facebook](https://facebook.com), [telegram](https://t.me), and [youtube](https://youtube.com).

Set the `effectiveDate` to the date your website launched live to enforce the `Privacy Policy` for your users.

```javascript
export const footerData = {
  networks: [
    {
      name: 'twitter',
      url: 'https://twitter.com/DarAsSahaba',
    },
  ],
  privacyPolicy: {
    effectiveDate: '18 July 2020',
  },
};
```

##### Contact Section

<img src="https://github.com/ragaeeb/das-js/blob/master/examples/contact.jpg" alt="Contact Section" />

Set your centre's contact information here. The email address will automatically be scrambled to be protected from spam bots but should be visible to the user.

If you do not want to display a specific attribute, simply omit it.

```javascript
export const contactInfo = {
  address: 'Your masjid address',

CHANGELOG.MD
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.8.5] - 2021-03-13

### Fixed

- [Update iqamah times for EST](https://github.com/ragaeeb/das-js/issues/388)

## [0.8.4] - 2021-02-21

### Fixed

- [Normalize NPE](https://github.com/ragaeeb/das-js/issues/355)
- [Guard against NPE for Object.keys](https://github.com/ragaeeb/das-js/issues/356)

### Updated

- [Update dependency node-sass to v5](https://github.com/ragaeeb/das-js/pull/266)
- [Update dependency @sentry/gatsby to v6.2.0](https://github.com/ragaeeb/das-js/pull/343)
- [Update gatsby monorepo](https://github.com/ragaeeb/das-js/pull/344)
- [Update dependency @babel/plugin-transform-runtime to v7.12.17](https://github.com/ragaeeb/das-js/pull/345)
- [Update dependency react-lazyload to v3.2.0](https://github.com/ragaeeb/das-js/pull/346)
- [Update dependency lint-staged to v10.5.4](https://github.com/ragaeeb/das-js/pull/347)
- [Update dependency react-icons to v4.2.0](https://github.com/ragaeeb/das-js/pull/348)
- [Update dependency eslint to v7.20.0](https://github.com/ragaeeb/das-js/pull/350)
- [Update dependency react-bootstrap to v1.5.0](https://github.com/ragaeeb/das-js/pull/351)
- [Update dependency eslint-config-prettier to v8](https://github.com/ragaeeb/das-js/pull/353)
- [Update dependency husky to v5.1.0](https://github.com/ragaeeb/das-js/pull/354)

## [0.8.3] - 2021-02-18

### Added

- [Add blog for reopening of musalla](https://github.com/ragaeeb/das-js/pull/352)

## [0.8.2] - 2021-02-01

### Updated

- [Update gatsby monorepo](https://github.com/ragaeeb/das-js/pull/334)
- [Update dependency preact to v10.5.12](https://github.com/ragaeeb/das-js/pull/336)
- [Update dependency eslint to v7.19.0](https://github.com/ragaeeb/das-js/pull/337)
- [Update dependency eslint-config-prettier to v7.2.0](https://github.com/ragaeeb/das-js/pull/338)
- [Update dependency @sentry/gatsby to v6](https://github.com/ragaeeb/das-js/pull/339)
- [Update dependency bootstrap to v4.6.0](https://github.com/ragaeeb/das-js/pull/340)
- [Update dependency husky to v5.0.8](https://github.com/ragaeeb/das-js/pull/342)

## [0.8.1] - 2021-01-04

### Updated

- [Update gatsby monorepo](https://github.com/ragaeeb/das-js/pull/329)
- [Update dependency gatsby-plugin-sass to v3](https://github.com/ragaeeb/das-js/pull/330)
- [Update dependency react-bootstrap to v1.4.3](https://github.com/ragaeeb/das-js/pull/331)

### Fixed

- [Omitting links key results in crash](https://github.com/ragaeeb/das-js/issues/332)

## [0.8.0] - 2021-01-04

### Added

- [Allow any prayer label to be linkable](https://github.com/ragaeeb/das-js/issues/326)

### Fixed

- [Privacy policy doesnt scale on mobile](https://github.com/ragaeeb/das-js/issues/318)

### Updated

- [Update dependency eslint to v7.17.0](https://github.com/ragaeeb/das-js/pull/319).
- [Update dependency preact to v10.5.9](https://github.com/ragaeeb/das-js/pull/321).
- [Update dependency eslint-plugin-prettier to v3.3.1](https://github.com/ragaeeb/das-js/pull/322).
- [Use official sentry plugin](https://github.com/ragaeeb/das-js/issues/323).
- [Disable cypress](https://github.com/ragaeeb/das-js/issues/324).
- [Seed the brand information from package.json](https://github.com/ragaeeb/das-js/issues/325).
- [Update README.MD](https://github.com/ragaeeb/das-js/issues/327).

## [0.7.0] - 2020-12-30

### Added

- [Add cypress](https://github.com/ragaeeb/das-js/issues/56)
- [Set up cypress runs on cypress.io](https://github.com/ragaeeb/das-js/issues/82)
- [Add chatwoot support](https://github.com/ragaeeb/das-js/issues/275)

### Updated

- [Refactor code to make it easier to adapt to any other masajid](https://github.com/ragaeeb/das-js/pull/312)
- [Update dependency gatsby to v2.29.3](https://github.com/ragaeeb/das-js/pull/314)
- [Update dependency eslint-plugin-react to v7.22.0](https://github.com/ragaeeb/das-js/pull/315)
- [Update dependency preact to v10.5.8](https://github.com/ragaeeb/das-js/pull/316)

## [0.6.12] - 2020-12-25

### Updated

- [Update dependency husky to v4.3.6](https://github.com/ragaeeb/das-js/pull/299)
- [Update gatsby monorepo](https://github.com/ragaeeb/das-js/pull/300)
- [Update dependency @babel/plugin-transform-runtime to v7.12.10](https://github.com/ragaeeb/das-js/pull/301)
- [Update dependency eslint-plugin-prettier to v3.3.0](https://github.com/ragaeeb/das-js/pull/303)
- [Update dependency eslint to v7.16.0](https://github.com/ragaeeb/das-js/pull/304)
- [Update dependency eslint-config-prettier to v7.1.0](https://github.com/ragaeeb/das-js/pull/305)
- [Bump node-notifier from 8.0.0 to 8.0.1](https://github.com/ragaeeb/das-js/pull/306)
- [Add new lockdown blog](https://github.com/ragaeeb/das-js/pull/308)

## [0.6.11] - 2020-12-06

### Updated

- [Update dependency eslint-plugin-prettier to v3.2.0](https://github.com/ragaeeb/das-js/pull/291)
- [Update dependency lint-staged to v10.5.3](https://github.com/ragaeeb/das-js/pull/292)
- [Update dependency eslint to v7.15.0](https://github.com/ragaeeb/das-js/pull/293)
- [Update dependency husky to v4.3.4](https://github.com/ragaeeb/das-js/pull/294)
- [Update fajr times](https://github.com/ragaeeb/das-js/pull/296)
- [Update dependency eslint-config-prettier to v7](https://github.com/ragaeeb/das-js/pull/297)

## [0.6.10] - 2020-12-02

### Updated

- [Update dependency prettier to v2.2.1](https://github.com/ragaeeb/das-js/pull/283)
- [Update dependency eslint to v7.14.0](https://github.com/ragaeeb/das-js/pull/284)
- [Update gatsby monorepo](https://github.com/ragaeeb/das-js/pull/285)
- [Update dependency lint-staged to v10.5.2](https://github.com/ragaeeb/das-js/pull/286)
- [Update dependency react-icons to v4](https://github.com/ragaeeb/das-js/pull/287)

src/mock/data.js
export const heroData = {
  calculation: {
    fajrAngle: 12,
    ishaAngle: 12,
    latitude: '45.3506',
    longitude: '-75.793',
    method: 'NauticalTwilight',
    timeZone: 'America/Toronto',
  },
  labels: {
    fajr: 'Fajr',
    sunrise: 'Sunrise',
    dhuhr: 'Dhuhr',
    asr: 'ʿAṣr',
    maghrib: 'Maġrib',
    isha: 'ʿIshāʾ',
    middleOfTheNight: '1/2 Night Begins',
    lastThirdOfTheNight: 'Last 1/3 Night Begins',
    jumuah: 'Khuṭbah al-Jumuʿah',
  },
  iqamahs: {
    fajr: {
      1: {
        1: '6:45 AM',
        21: '6:35 AM',
      },
      2: {
        1: '6:30 AM',
        11: '6:15 AM',
        21: '6:05 AM',
      },
      3: {
        1: '5:50 AM',
        9: '5:35 AM',
        14: '6:20 AM',
        22: '6:10 AM',
      },
      4: {
        1: '5:50 AM',
        11: '5:30 AM',
        21: '5:10 AM',
      },
      5: {
        1: '4:45 AM',
        11: '4:30 AM',
        21: '4:20 AM',
      },
      6: '4:10 AM',
      7: {
        1: '4:15 AM',
        11: '4:25 AM',
        21: '4:40 AM',
      },
      8: {
        1: '5:00 AM',
        11: '5:15 AM',
        21: '5:30 AM',
      },
      9: {
        1: '5:45 AM',
        11: '5:55 AM',
        21: '6:05 AM',
      },
      10: {
        1: '6:15 AM',
        11: '6:30 AM',
        21: '6:40 AM',
      },
      11: {
        1: '6:00 AM',
        11: '6:10 AM',
        21: '6:20 AM',
      },
      12: {
        1: '6:35 AM',
        11: '6:45 AM',
      },
    },
    dhuhr: {
      1: '12:30 PM',
      2: '12:30 PM',
      3: {
        1: '12:30 PM',
        14: '1:30 PM',
      },
      4: '1:30 PM',
      5: '1:30 PM',
      6: '1:30 PM',
      7: '1:30 PM',
      8: '1:30 PM',
      9: '1:30 PM',
      10: '1:30 PM',
      11: '12:30 PM',
      12: '12:30 PM',
    },
    jumuah: {
      1: ['12:15 PM', '1:00 PM'],
      2: '12:15 PM',
      3: {
        1: '12:15 PM',
        14: '1:00 PM',
      },
      4: ['1:00 PM', '1:45 PM'],
      5: ['1:00 PM', '1:45 PM'],
      6: ['1:00 PM', '1:45 PM'],
      7: ['1:00 PM', '1:45 PM'],
      8: ['1:00 PM', '1:45 PM'],
      9: ['1:00 PM', '1:45 PM'],
      10: ['1:00 PM', '1:45 PM'],
      11: ['12:15 PM', '1:00 PM'],
      12: ['12:15 PM', '1:00 PM'],
    },
    asr: '+20 mins',
    maghrib: '+5 mins',
    isha: {
      1: '7:30 PM',
      2: '7:30 PM',
      3: {
        1: '7:30 PM',
        14: '8:45 PM',
        22: '9:00 PM',
      },
      4: {
        1: '9:15 PM',
        11: '9:30 PM',
        21: '9:45 PM',
      },
      5: {
        1: '10:00 PM',
        11: '10:15 PM',
        21: '10:30 PM',
      },
      6: {
        1: '10:30 PM',
        11: '10:40 PM',
        21: '10:40 PM',
      },
      7: {
        1: '10:30 PM',
        11: '10:20 PM',
        21: '10:10 PM',
      },
      8: {
        1: '10:00 PM',
        11: '9:45 PM',
        21: '9:30 PM',
      },
      9: {
        1: '9:15 PM',
        11: '9:00 PM',
        21: '8:45 PM',
      },
      10: {
        1: '8:30 PM',
        11: '8:15 PM',
        21: '8:00 PM',
      },
      11: '7:30 PM',
      12: '7:30 PM',
    },
  },
  hijriAdjust: -1,
  links: {
    fajr: 'https://archive.org/download/fajr-letter-das/Letter_Fajr-Ishaa_Ottawa-2016-06-27.pdf',
  },
  calendarUrl:
    'https://calendar.google.com/calendar/embed?src=60me58mrktt0lt24mijjkhddvc%40group.calendar.google.com&ctz=America%2FNew_York',
  istijabaText:
    'The Messenger of Allāh ﷺ said, “Friday has twelve meaning hours to it. There is no Muslim who asks Allāh for anything (during it) except that Allāh gives it to him so seek it during the last hour after ʿaṣr.” [Abū Dāwūd al-Sijistānī, al-Sunan #1048] [Al-Albānī: Ṣaḥīḥ]',
};

export const aboutData = {
  paragraphOne:
    'As-Ṣaḥāba Muṣallá was officially opened in October 2005 and has been serving the Ramsey community as a place for Muslims to pray 5 times a day, (including the Jumʿuah Prayer) ever since. Lectures on Islām are held on a regular basis and the presence of the Muṣallá has affected the Ramsey area in a positive way, walḥamdulillāh.',
  paragraphTwo: 'As a service to the Muslim community, Dār As-Ṣaḥāba Provides the Following:',
  services: [
    {
      title: 'Five Daily Ṣalawāt',
      body:
        'The muṣallá is open to allow Muslims (both brothers and sisters) a location to pray their five daily ṣalawāt.',
    },
    {
      title: 'Ṣalat al-Jumʿuah',
      body:
        'Every week, salat-ul-Jumʿuah is held at Dār As-Ṣaḥāba (due to limited space, for brothers only), where the khaṭīb deliver the khuṭbah strictly adhering to the Qur’ān & the Sunnah.',
    },
    {
      title: 'Ṣalat al-Tarawīḥ',
      body:
        'During the month of Ramaḍān, every evening Ṣalat al-Tarawīḥ is held (due to limited space, for brothers only).',
    },
    {
      title: 'Ṣalat al-ʿĪd',
      body:
        'For both ʿĪd al-Fiṭr and ʿĪd al-Aḍḥá, Dār As-Ṣaḥāba holds ʿĪd prayers for the community at a neighbouring park, weather permitting, or community centre.',
    },
    {
      title: 'Lectures',
      body:
        'A few knowledgeable brothers remind us of the importance of various facets of our dīn in weekly gatherings. Occasionally, tele-lectures of our esteemed scholars are streamed via the internet at the muṣallá for all to hear.',
    },
    {
      title: 'Distributing reading and audio/visual materials',
      body:
        'To those who are interested about Islām, we have provided copies of the Qur’an with translations in the English and French languages; for non-Arabic speaking Muslims, we have also provided translated copies of the Qur’ān in other languages such as Spanish, Somali, Farsi, Urdu, Pashto, etc. As well, we have provided books and tapes/CDs for people ranging from those interested in Islām to those that are interested in further pursuing Islām academically.',
    },
  ],
};

export const donationsData = {
  directDeposit:
    'If you would like your monthly donation amount to be automatically deducted from your bank account, we have a direct-deposit form available that you can fill out. Simply drop by the muṣallá when any of the administration is there and ask for the form so that we can get you set up in shā’ Allah.',
  etransfer: 'dar.as.sahaba@hotmail.com',
  expenses: [
    { label: 'Rent', value: 3400 },
    { label: 'Utilities', value: 600 },
    { label: 'Cleaning', value: 240 },
    { label: 'Phone/Internet', value: 117 },
    { label: 'Supplies', value: 160 },
  ],
  khutbah:
    'After the Friday sermon, donations for the muṣallá are collected. Sometimes there are baked goods and food that are also being sold for fundraising for the muṣallá. Please donate generously for the sake of Allāh.',
  paypal:
    'https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=M36T4MXQHLUHJ&source=url',
  reminderDawah:
    'When you participate in online giving, you’re also helping to streamline administrative overhead costs – thus, putting more of your dollars to work in keeping the muṣallá running.',
  reminderReward:
    'For as long as the muṣallá remains open by your donations, you are collecting your reward for every person who is establishing the prayer, for every lesson that is being organized and every Muslim who is learning from the lesson, from every lost youth seeking the truth and changes their life around by way of it. You are collecting your reward for everyone that is affected by your good deeds.',
};

export const contactInfo = {
  cta:
    'Feel free to drop by the muṣallá at any time to perform the congregational prayers and to partake in the activities. If you would require help with learning about Islām and wish to speak to one of the members of the administration, please contact us through the website first so that we can schedule an appointment with you at a convenient time in shā’ Allāh.',
  emailButton: 'Email Us',
  mailingList: 'Mailing List',
  email: 'dar.as.sahaba@gmail.com',
  mapUrl: 'https://goo.gl/maps/14LYnHa7R9sZsQBG7',
  telButton: 'Call Us',
  tel: '+16133215459',
  address: '2835 Dumaurier Ave, Ottawa, ON, Canada. K2B 7W3',
};

export const footerData = {
  networks: [
    {
      name: 'twitter',
      url: 'https://twitter.com/DarAsSahaba',
    },
    {
      name: 'soundcloud',
      url: 'https://soundcloud.com/DarAsSahaba',
    },
    {
      name: 'instagram',
      url: 'https://instagram.com/DarAsSahaba',
    },
    {
      name: 'facebook',
      url: 'https://facebook.com/DarAsSahaba',
    },
    {
      name: 'telegram',
      url: 'https://t.me/DarAsSahaba',
    },
    {
      name: 'youtube',
      url: 'https://www.youtube.com/channel/UC4JRttylAMmZubEmFOjX7Vg',
    },
  ],
  privacyPolicy: {
    effectiveDate: '18 July 2020',
  },
};
